### DISCUSSION TITLE
Cinnamon option to look like a dock?

### DISCUSSION THREAD

**Discussion by eugenialoli** [linuxmint/discussions#418] [Ideas]
I'd love for a Dock mode, that auto-resizes. Basically, when such an option is selected, the panel becomes a mini-panel (like a dock), centered in the screen, that auto-resizes when more icons are added.

Also, please redesign a bit the panel settings window, it doesn't fit on 1366x768 screens (I have 4 such laptops...). Thx!

**Comment by 100savage** [comment]
To tweak your panel settings take a look at this https://github.com/orgs/linuxmint/discussions/374

**Reply by eugenialoli** [reply]
I don't see how this helps me. This is about the system settings window, not about the cinnamon panel.

**Reply by Hilyxx** [reply]
Plank is a good alternative for users that want a dock. This app is light and is able to auto-resize when more icons are added.

**Reply by eugenialoli** [reply]
Plank is not stable. I just need Cinnamon to auto-resize. I don't think that this is too difficult for the devs to implement, and it would add dock functionality to the OS, which is a feature that a lot of people are after.

**Reply by ChrisHonegger** [reply]
I've been using Plank for quite a while and I like it. However, there is no Wayland support. Therefore, it is no future option.

**Reply by leonsk29** [reply]
Plank is unmaintained, people need to move to Plank Reloaded (https://github.com/zquestz/plank-reloaded). But others are right, having this functionality natively would be nice, since Wayland is coming and no dock is Wayland-compatible yet.

Oh, and native blurred transparency would be nice, too, like COSMIC's "frosted glass" effect.

**Comment by 100savage** [comment]
It also works for the systems settings. I'm only trying to help you and you come off as rude to me. If you don't want suggestions don't ask for help.

**Reply by eugenialoli** [reply]
I didn't actually ask for a hack that gets overwritten every time you update cinnamon. I asked for an officially supported feature, that has a gui in the panel window. I want this to exist for all users, not just a one-time thing for me. It's a feature request, not a question.

I came to this page after following the links of Linux Mint's github page (from their bug report page) for "feature request". If this is supposed to be a forum instead, I apologize, but I was lead to believe that these are the feature request pages.

**Reply by Jopp-gh** [reply]
Well for new users (I include myself) Mint isn't exactly the same as the commercial OS's we were exposed for years.

I think suggestions / feature requests are always helpful. On the other side, there are more experienced users who try to help and maybe have great alternatives.
These helping people shouldn't be seen an obstacles that try to downvote feature requests.

Mint is continually improving. Still, developers have priorities what's really important to implement. Also, coding isn't simple and requires context-aware knowledge.

I also had a nice discussion because I wanted and still want a workspace related dock. Example, to show in workplace "office" just  office apps and nothing else, like online apps, etc. Meanwhile I use workspaces and open apps with the cinnamon main panel, some few apps are in my panel. This works for me right now and my feature req isn't my main concern anymore. If the developers have time and see the value of my request it will be implemented.

After a initial period of confusion I feel now pretty much at home on Mint

**Reply by eugenialoli** [reply]
I've been using Linux since 1998. I simply wanted to post a feature request. As for Mint continually improving, it is, but the actual Cinnamon panel received only a handful of commits in the last year here on github. It's not exactly thriving or expanded in any way. It's now considered "it's what it is". That's stagnation in my opinion.

**Reply by Hobson318** [reply]
I would like to see this issue raised again. To add the ability to resize the panel's horizontal width. I use an ultra-wide and have moved the applets to the center but the system icons are still way over in the corner. ZorinOs has this option built in. So the code to do this exists already. I don't want to hide the panel and use a dock to fix an issue that shouldn't be hard to implement. It's being done elsewhere. I know the devs have priorities but this should be an easy one since mint and zorin aren't that far apart from each other. I just don't know how to do it. Please and thank you.

**Reply by Jopp-gh** [reply]
The solution to get a MacOs dock on Mint is very easy to archive and doesn't need any plugin.

1. Open: settings>panel
2. put your preferred apps at the center in the bottom panel
3. adjust icon size for your bottom panel
4. set your panel to hide intelligently or how it's called. Done. 

Bonus: there's even a spice to render panels with a transparent background. 

So much fuss for such a little feature, makes me perplexed. 🤷‍♂️

**Reply by eugenialoli** [reply]
What you just suggested doesn't do what we ask.

**Reply by Jopp-gh** [reply]
>3. adjust icon size for your bottom panel

Well, I think to find a middle ground size for icons shouldn't be that hard. 

Anyway, this request remembers me the analogy of infinite tabs in browsers: for me, just a mess and struggle to read. I keep very little inactive apps in my Dock by default. 

Have you tried to use ULauncher or similar aids instead?

**Comment by ghost** [comment]
I think the best way to implement this is to have "floating panels". If you go to panel edit mode and add a panel, you can only add panels that stretch along one side of the screen all the way to the end. A "floating" panel could be pinned to the middle or side of an edge, and be designed to be "elastic" where it grows and shrinks depending on the contents. I could foresee it being easier to still only have one panel per edge.

I think you can use the existing panel system, just have a setting where you can set it to span, center align, left/top align, or right/bottom align

Attached is an example of two floating docks, one pinned to the center of the bottom edge for apps, and one pinned to the right of the top edge for the utility icons. both would be treated as auto-hide or intelligently hide
![desktop](https://github.com/user-attachments/assets/56d4ceca-44ea-4144-ab36-11909bb6328e)

**Reply by utnapishtim2026** [reply]
I think this is whaat you ar looking for: https://www.gnome-look.org/p/2366026/  

I have created it myself.

<img width="1366" height="768" alt="29" src="https://github.com/user-attachments/assets/7a526973-820a-4860-8da2-61c5968d5fc8" />
<img width="1366" height="768" alt="32" src="https://github.com/user-attachments/assets/dd974f7c-d5cb-4f65-8dde-388ecd9bc1c2" />
<img width="1366" height="768" alt="61" src="https://github.com/user-attachments/assets/43560fe1-a937-485e-81d3-8f2b81dfb520" />
<img width="1366" height="768" alt="91" src="https://github.com/user-attachments/assets/7a594b26-79f9-4aaf-82f8-fd8d63b119b7" />
<img width="1366" height="768" alt="116" src="https://github.com/user-attachments/assets/a26953d6-3237-4436-bcad-15dcbc742916" />

**Comment by BeatLink** [comment]
KDE has a really good implementation of this.

**Comment by BeatLink** [comment]
Now that Cinnamon is becoming more stable on Wayland, this issue is growing more important. The wayland architecture means that certain variables and functions that docks rely on (such as window size and positioning) are only available to the compositor. Its why plank-dock isnt able to be supported on wayland. 

The functionality is mostly there in Cinnamon with panel autohide and the left, center and right zones for separating applets. All that's needed is some CSS to make the panel transparent in areas where there aren't any applets.

**Comment by agentl074** [comment]
I second a native dock option. For me, Plank had issues, and while Cairodock works fine, a native option would create less fuss. Maybe a dock mode that extends the existing panel -- much like MacOS or Gnome does this?

**Comment by BeatLink** [comment]
The Centered Cinnamon Dock extension is a working proof of concept for this. It would be great if this could be implemented into cinnamon's core code and given some development support. The extension is 90% there. Just some bugs and graphical glitches to work out. 

https://cinnamon-spices.linuxmint.com/extensions/view/110

**Comment by claudiux** [comment]
There are two “dock” style themes:

- [SimpleDock](https://cinnamon-spices.linuxmint.com/themes/view/SimpleDock)
- [OrchideaDock](https://cinnamon-spices.linuxmint.com/themes/view/OrchideaDock)

**Reply by leonsk29** [reply]
> There are two “dock” style themes:
> 
> * [SimpleDock](https://cinnamon-spices.linuxmint.com/themes/view/SimpleDock)
> * [OrchideaDock](https://cinnamon-spices.linuxmint.com/themes/view/OrchideaDock)

Those are too old by now. One is 2 years old and the other is 1. They seem unmaintained.

**Reply by claudiux** [reply]
Ping @brownsr @iamcheyan

**Reply by iamcheyan** [reply]
> * OrchideaDock

Actually, implementing a dock in Cinnamon is entirely possible through CSS. OrchideaDock achieves this by hollowing out the side styles to mimic a dock layout. Despite being over two years old, the theme still functions flawlessly on the newest Linux Mint—definitely worth trying.

**Reply by BeatLink** [reply]
This just goes to show how easy implementation is theoretically. This should be a core part of cinnamon.

**Reply by utnapishtim2026** [reply]
you mean something like this one:  https://www.gnome-look.org/p/2366026/
](https://www.gnome-look.org/p/2366026/)<img width="1366" height="768" alt="1" src="https://github.com/user-attachments/assets/a02b2903-caf2-47f8-879c-d56322e231d9" />
<img width="1366" height="768" alt="2" src="https://github.com/user-attachments/assets/f2a1d9b2-d5ad-46c4-a626-9a8345bba372" />
<img width="1366" height="768" alt="3" src="https://github.com/user-attachments/assets/a52f6e57-5285-4ce7-9c6c-d39e8dcb08b1" />

**Comment by Hilyxx** [comment]
I’ve developed a new dock for Cinnamon: Mint-Dock (currently under review ). It’s a native dock that is fully integrated into the desktop.

If you’re familiar with Plank or Dash to Dock on GNOME Shell, you should feel right at home, as Mint-Dock borrows many of the features from both docks and adds even more. It is also compatible with both X11 and Wayland.

Some of its features include:
- Custom background color
- Rearrange dock icons and drag and drop applications from the Cinnamon menu onto the dock
- Choose the dock position on the screen (top, bottom, left, right, start, center, or end)
- Two hiding modes: smart hiding (Intellihide) and automatic hiding
- Several configurable options: show tooltips, show separators, show the trash, and more
- Notification badges and progress bars are displayed

The dock has gone through a very thorough debugging and testing phase, so it should be very stable and have no noticeable impact on Cinnamon’s performance. That said, you’re never completely safe from a last-minute bug...

By default, the dock is positioned on the left, since the Cinnamon panel is at the bottom on a fresh installation. Of course, you can place it wherever you like.

Simply right-click on the edges of the dock to access its settings.

Enjoy !

<img width="2600" height="1445" alt="screenshot" src="https://github.com/user-attachments/assets/b6700a9a-96d4-4a87-9938-b2913276e6b4" />

**Reply by BeatLink** [reply]
This looks clean! I like it. Does it work on Wayland? That's the main reason i had to move from plank reloaded

**Reply by Hilyxx** [reply]
Yes, I’ve mentioned it: it works on both X11 and Wayland. Since I can’t test Wayland directly on my machine (I get a black screen on LMDE 7), I tested my dock on a virtual machine (with LM23 Alfa) and didn’t encounter any particular issues.

That said, with Wayland, we’ll really have to see how things go once Linux Mint 23 is released, as it should no longer be considered experimental at that point. The team could still change a few things between now and December, though...

**Reply by BeatLink** [reply]
nice!

**Reply by Axel-Erfurt** [reply]
One Problem, 'Pin to dock' changes /org/cinnamon/favorite-apps, so it changes the cinnamon menu too.

**Reply by Hilyxx** [reply]
Indeed, the dock uses the same global.settings.get_strv key as the Cinnamon menu to add an application to the favorites and therefore pin it to the dock. How does this cause a problem?

**Reply by eugenialoli** [reply]
Where can I download a test version? Was it written with Claude?

**Reply by Hilyxx** [reply]
You can test the dock by downloading the ZIP file from my Git repository:
https://github.com/Hilyxx/cinnamon-spices-extensions
And no, I didn’t use Claude to write my extension. That said, I did use AI to help fix bugs.

**Reply by eugenialoli** [reply]
I'm confused. There a gazillion extensions there, which one is the one with the Dock? I only see a Screenshot one that carries your username.

**Reply by Hilyxx** [reply]
If you couldn’t find it, that’s normal,I didn’t give you the right link. Sorry!
https://github.com/Hilyxx/cinnamon-spices-extensions/tree/mint-dock

I had time to fix some of the issues. Now we'll have to wait for a maintainer to approve my PR.

**Comment by Axel-Erfurt** [comment]
It makes the favorites list longer in menu. It might be better if Mint-Dock saved its own list.

**Reply by Hilyxx** [reply]
That’s a side effect I hadn’t thought of. It’s not really an issue for me, but for people with a lot of pinned apps, it could become a problem. Then again, you can always just hide the menu sidebar lol. I’ll look into it.

**Reply by Axel-Erfurt** [reply]
Good tip! You're right, I don't actually need the sidebar in menu.

**Reply by Hilyxx** [reply]
I changed how pinned apps work. I'm no longer using the favorite-apps key, but a custom array instead. It's way more convenient and won't mess with the Cinnamon menu anymore. Other users will probably ask the same question, so we might as well change it now.
